### PR TITLE
PR 4: migrate forms router to shared form service

### PR DESCRIPTION
**Pull request by lotorvik**
## Summary
- replace fyllut-backend's legacy FormService with shared formService in forms routes and renderIndex
- remove the deleted local form-loading helpers and update tests/mocks for the new forms-api-backed behavior
- fix shared form mapping/client edge cases found during review, including select forwarding and production-form fallbacks

## Testing
- pnpm --filter @navikt/skjemadigitalisering-shared-domain exec vitest run src/utils/forms-api-backwards-compatibility/form.test.ts
- pnpm --filter @navikt/skjemadigitalisering-shared-domain run check-types
- pnpm --filter @navikt/skjemadigitalisering-shared-backend exec vitest run src/services/form/formClient.test.ts
- pnpm --filter @navikt/skjemadigitalisering-shared-backend run check-types
- pnpm --filter @navikt/fyllut-backend exec vitest run src/routers/api/forms/form.test.ts
- pnpm --filter @navikt/fyllut-backend run check-types
- pnpm --filter @navikt/fyllut-backend run test

### CODE REVIEW

File: packages/fyllut-backend/src/routers/api/forms/form.test.ts
Comment (akgagnat) [inline packages/fyllut-backend/src/routers/api/forms/form.test.ts:1]: slett og verifiser testdekning annet sted

File: packages/shared-domain/src/utils/forms-api-backwards-compatibility/form.ts [hunk 1]
```diff
@@ -23,6 +23,7 @@ const mapNavFormToForm = (form: NavFormType): Form => {
   const {
     id,
     revision,
+    modified,
```
Comment (akgagnat) [inline packages/shared-domain/src/utils/forms-api-backwards-compatibility/form.ts]: dropp denne? og bruk changedAt på fyllut index-siden

**Review by akgagnat — APPROVED** [review]
👍 💯
